text/rocq-roadmap.md

```diff
@@ -5,14 +5,14 @@ Rocq Long Term Roadmap
 
 The Rocq Development Team, building on 40 years of experience with Coq, is pleased to announce the Rocq Prover, the next-generation proof assistant. The Rocq Prover is designed to empower a diverse range of users, from mathematicians seeking formal rigor to software engineers building high-assurance systems.
 
-While Rocq leverages the rich heritage of Coq, it embraces a more agile development philosophy. This allows us to deliver impactful changes more frequently, accelerating innovation and user experience improvements.
+While Rocq leverages the rich heritage of Coq, it aims to keep evolving: deliver impactful changes, enable new research and improve the user experience.
```

Comment:
I understand that "embraces a more agile development philosophy" seems like marketing bullshit without anything behind, but I don't like the proposed replacement "aims to keep evolving" which is almost tautological. According to the second law of software evolution, software must keep evolving to adapt to its context, or it becomes less and less useful. No need for a roadmap or vision document to state that...

It may not be completely clear in this introduction what we mean by "more agile development philosophy", but it is related to section 3 of this document in particular.

Reply:
My understanding of the paragraph is to dispel the impression that Coq is a previous-generation system that has lost its velocity and will forever remain similar to how it is today. So the idea is to mention age/history in a positive light, yet clarify that change is still welcome. "Move fast and break things", as Facebook used to say. How would you suggest saying this? (Or maybe you definitely think that it's not worth addressing this concern?)

Reply:
If I may develop, the meaning of this sentence is, in a way, to contrast with other proof assistants' own development methodology:

- We have a rich history and a rich ecosystem (richer and longer than most other systems) and part of the philosophy of Rocq will be to keep *leveraging this heritage* (Jim noted that this sounds like marketing, but I am not sure what would be a more appropriate word here).
- Yet, we know that we are still far from a state where we can keep delivering important improvements to the activity of deductive verification without making significant changes (breaking changes) to the system, and we aim to keep accelerating to deliver this important improvements frequently.
- This means finding the delicate balance between breaking changes / development pace on the one hand, and maintenance of the ecosystem on the other.

Reply:
This sentence

> We have a rich history and a rich ecosystem

makes me think that, as part of this history, we could emphasize our strengths in proof theory and in formalization techniques. This could be said e.g. in the second sentence:

The Rocq Prover prioritizes self-verification and performance optimization, relying on a solid tradition of research in logical foundations and formalization techniques to ensure the highest level of trust and efficiency in your formal proofs.

Reply:
Maybe also adding a word on our background on programming (thus justifying our ability to take care of "efficient"). E.g.:

The Rocq Prover prioritizes self-verification and performance optimization, relying on a solid tradition of research in logical foundations, formalization techniques and programming to ensure the highest level of trust and efficiency in your formal proofs.

Reply:
IHMO in the initial phrasing `embraces a more agile development philosophy`: I believe that `agile` could/should be kept, given it faithfully summarizes Rocq's workflow in terms of project management (cf. e.g. the development and use of a dedicated DevOps tooling, or the weekly Coq calls that might be seen as agile ceremonies); so if something should be changed in this initial phrasing, maybe rather the word `philosophy`(?)